validator/client/src/main/java/tech/pegasys/teku/validator/client/doppelganger/DoppelgangerDetector.java

```diff
@@ -181,9 +184,14 @@ private SafeFuture<Void> performDoppelgangerCheck(final Set<BLSPublicKey> pubKey
                       mapToAbbreviatedKeys(pubKeys).collect(Collectors.toSet()));
                   return stopDoppelgangerDetectorTask(new HashMap<>());
                 } else {
-                  final UInt64 previousEpoch =
-                      currentEpoch.isZero() ? currentEpoch : currentEpoch.minus(UInt64.ONE);
-                  return checkDoppelgangersAtEpoch(pubKeys, currentSlot, previousEpoch);
+                  if (firstCheck.get()) {
+                    firstCheck.set(false);
```

Comment:
needs to be an atomic check and set - `compareAndSet`

Reply:
Done